cmd/pulumi-test-language/tests.go

```diff
@@ -869,6 +869,76 @@ var languageTests = map[string]languageTest{
 			},
 		},
 	},
+	"l2-invoke-secrets": {
+		providers: []plugin.Provider{
+			&providers.SimpleInvokeProvider{},
+			&providers.SimpleProvider{},
+		},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, err error,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					requireStackResource(l, err, changes)
+					var stack *resource.State
+					for _, r := range snap.Resources {
+						if r.Type == resource.RootStackType {
+							stack = r
+							break
+						}
+					}
+
+					require.NotNil(l, stack, "expected a stack resource")
+
+					outputs := stack.Outputs
+					assertPropertyMapMember(l, outputs, "nonSecret",
+						resource.NewStringProperty("hello world"))
+					assertPropertyMapMember(l, outputs, "firstSecret",
+						resource.MakeSecret(resource.NewStringProperty("hello world")))
+					assertPropertyMapMember(l, outputs, "secondSecret",
+						resource.MakeSecret(resource.NewStringProperty("goodbye world")))
+				},
+			},
+		},
+	},
+	"l2-invoke-dependencies": {
+		providers: []plugin.Provider{
+			&providers.SimpleInvokeProvider{},
+			&providers.SimpleProvider{},
+		},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, err error,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					requireStackResource(l, err, changes)
+					var first *resource.State
+					var second *resource.State
+					for _, r := range snap.Resources {
+						if r.URN.Name() == "first" {
+							first = r
+						}
+						if r.URN.Name() == "second" {
+							second = r
+						}
+					}
+
+					require.NotNil(l, first, "expected first resource")
+					require.NotNil(l, second, "expected second resource")
+					require.Empty(l, first.Dependencies, "expected no dependencies")
+					require.Len(l, second.Dependencies, 1, "expected one dependency")
+					dependencies, ok := second.PropertyDependencies["value"]
+					require.True(l, ok, "expected dependency on property 'value'")
+					require.Len(l, dependencies, 1, "expected one dependency")
+					require.Equal(l, first.URN, dependencies[0], "expected second to depend on first")
+					require.Len(l, second.Dependencies, 1, "second has 1 direct dependency")
```

Comment:
Nit: isn't this the same assert as line 931?

Reply:
Nice catch! removed the double check